src/RpcClient.ts

```diff
@@ -72,58 +85,241 @@ export class RpcClient {
+   * @returns {Promise<getAssetResponseList>}
+   */
+  async getAssetsByGroup(params: AssetsByGroup): Promise<getAssetResponseList> {
+    const url = `${this.connection.rpcEndpoint}`;
+    const response = await fetch(url, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        jsonrpc: "2.0",
+        id: "my-id",
+        method: "getAssetsByGroup",
+        params: params,
+      }),
+    });
+    console.log(params);
+    const data = await response.json();
+    return data.result;
+  }
+  /**
+   * Request an allocation of lamports to the specified address
+   * @returns {Promise<getAssetResponseList>}
+   */
+  async getAssetsByOwner(params: AssetsByOwner): Promise<getAssetResponseList> {
+    const url = `${this.connection.rpcEndpoint}`;
+    const response = await fetch(url, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        jsonrpc: "2.0",
+        id: "my-id",
+        method: "getAssetsByOwner",
+        params: params,
+      }),
+    });
+    const data = await response.json();
+    return data.result;
+  }
+  /**
+   * Request an allocation of lamports to the specified address
+   * @returns {Promise<getAssetResponseList>}
+   */
+  async getAssetsByCreator(
+    params: AssetsByCreator
+  ): Promise<getAssetResponseList> {
+    const url = `${this.connection.rpcEndpoint}`;
+    const response = await fetch(url, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        jsonrpc: "2.0",
+        id: "my-id",
+        method: "getAssetsByCreator",
+        params: params,
+      }),
+    });
+    const data = await response.json();
+    return data.result;
+  }
+  /**
+   * Request an allocation of lamports to the specified address
+   * @returns {Promise<getAssetResponseList>}
+   */
+
+  async getAssetsByAuthority(
+    params: AssetsByAuthority
+  ): Promise<getAssetResponseList> {
+    const url = `${this.connection.rpcEndpoint}`;
+    const response = await fetch(url, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        jsonrpc: "2.0",
+        id: "my-id",
+        method: "getAssetsByAuthority",
+        params: params,
+      }),
+    });
+    const data = await response.json();
+    return data.result;
+  }
+  /**
+   * Request an allocation of lamports to the specified address
+   * @returns {Promise<getAssetResponseList>}
+   */
+  async searchAssets(params: SearchAssets): Promise<getAssetResponseList> {
+    const url = `${this.connection.rpcEndpoint}`;
+    const response = await fetch(url, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        jsonrpc: "2.0",
+        id: "my-id",
```

Comment:
Can the ID please be an option to configure per client? E.g. you can create a helius client with `my-fun-app` as the id. We see the ID when we look at requests. 

Batch requests should have unique IDs, so you can make them `my-fun-app-1`, `my-fun-app-2`, etc.

If no ID is provided make it `helius-sdk`